interfaces/acados_template/acados_template/acados_casadi_ocp_solver.py

```diff
@@ -572,14 +784,35 @@ def get(self, stage: int, field: str):
                 g_lam = self.nlp_sol_lam_g[self.index_map['lam_gnl_in_lam_g'][stage]]
 
             lbx_lam = np.maximum(0, -bx_lam)
-            lbu_lam = np.maximum(0, -bu_lam)
-            lg_lam = np.maximum(0, -g_lam)
             ubx_lam = np.maximum(0, bx_lam)
+            lbu_lam = np.maximum(0, -bu_lam)
             ubu_lam = np.maximum(0, bu_lam)
-            ug_lam = np.maximum(0, g_lam)
-            lam = np.concatenate((lbu_lam, lbx_lam, lg_lam, ubu_lam, ubx_lam, ug_lam))
+            if any([dims.ns_0, dims.ns, dims.ns_e]):
+                lw_soft_lam = self.nlp_sol_lam_x[self.index_map['sl_in_w'][stage]]
+                uw_soft_lam = self.nlp_sol_lam_x[self.index_map['su_in_w'][stage]]
+                lg_soft_lam = self.nlp_sol_lam_g[self.index_map['lam_sl_in_lam_g'][stage]]
+                ug_soft_lam = self.nlp_sol_lam_g[self.index_map['lam_su_in_lam_g'][stage]]
+                # TODO: could be improved?
+                if self.index_map['lam_gnl_in_lam_g'][stage]:
+                    if self.index_map['lam_gnl_in_lam_g'][stage] < self.index_map['lam_su_in_lam_g'][stage]: 
+                        # hard constraints before soft constraints
```

Comment:
Please add an exception to handle any implcit requirements.

Also add a TODO in the comment.

It should be something like this:
```
lbg_lam = np.zeros(..)
lbg[offset+idxsh] = lg_soft_lam
```
I think it also makes sense to have the not slacked constraints index vectors, like idxnsh available.